include/sgx/faasm_sgx_attestation.h

```diff
@@ -84,13 +79,57 @@ typedef struct __attribute((packed)) _msg_hash_fct
     uint8_t fct_name[];
 } sgx_wamr_msg_hash_fct_t;
 
+typedef struct __attribute((packed)) _msg_state_read
+{
+    sgx_wamr_msg_hdr_t hdr;
+    uint8_t state_name[];
+} sgx_wamr_msg_state_read_t;
+
+typedef struct __attribute((packed)) _msg_state_read_res
+{
+    sgx_wamr_msg_hdr_t hdr;
+    uint8_t state_secret[SGX_AESGCM_KEY_SIZE];
+    uint8_t buffer_nonce[SGX_AESGCM_IV_SIZE];
+    uint8_t data[];
+} sgx_wamr_msg_state_read_res_t;
+
+typedef struct __attribute((packed)) _msg_state_write
+{
+    sgx_wamr_msg_hdr_t hdr;
+    uint8_t buffer_nonce[SGX_AESGCM_IV_SIZE];
+    uint32_t name_length;
+    uint8_t data[];
+} sgx_wamr_msg_state_write_t;
+
 typedef struct _faaslet_sgx_msg_buffer
 {
     uint32_t buffer_len;
     sgx_wamr_msg_t* buffer_ptr;
 } faaslet_sgx_msg_buffer_t;
 
-typedef struct _attestation_env
+typedef struct _faaslet_sgx_gp_buffer
+{
+    uint32_t buffer_len;
+    void* buffer_ptr;
+} faaslet_sgx_gp_buffer_t;
```

Comment:
Some of these structs seem to be recreating standard library objects. Why can't we use `std::vector`s?